tools/o3de-doxygen/index.md

```diff
@@ -0,0 +1,7 @@
+
+    Welcome to the **Open 3D Engine (O3DE)** API Reference for the **WhiteBox Gem**!
```

Comment:
"For the WhiteBox Gem"??

Reply:
This is a file that the script changes. (White Box Gem is the last one that it generated for.) I think I will add this index.md to the ignore list (and name it something better, like "temp"). And then remove it from this PR.

Reply:
I've added a clean up step in o3de-*-api.sh that removes the temp file